src/backend/commands/tablecmds.c

```diff
@@ -6504,6 +6504,8 @@ ATRewriteTable(AlteredTableInfo *tab, Oid OIDNewHeap, LOCKMODE lockmode)
 
 		econtext = GetPerTupleExprContext(estate);
 
+		Insist(newTupDesc && oldTupDesc);
```

Comment:
1) What does newTupDesc have to do with it?
2) Comparing oldTupDesc with NULL below seems to hint...

Reply:
1. It doesn't make sense to check oldTupDesc without checking newTupDesc
2. It's either changed in that branch so it can be NULL or just pointless because both newTupDesc and oldTupDesc are dereferenced rigth in `MakeSingleTupleTableSlot` and in https://github.com/arenadata/gpdb/blob/137ca22444525f22ef7d3c9518ff68b4a3ae8845/src/backend/commands/tablecmds.c#L6518

Reply:
Then I think that the [condition](https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/commands/tablecmds.c#L6801) should be replaced with an assertion and the [dead code](https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/commands/tablecmds.c#L6806-L6807) should be removed.

Reply:
Done

Reply:
I do not see assert, and I still see insist.

Reply:
Switched to Assert